app/(main)/o/[slug]/conversation-history.tsx

```diff
@@ -15,48 +20,256 @@ interface Props {
 
 type Conversation = z.infer<typeof conversationSchema>;
 
+type ConversationsByDate = {
+  today: Conversation[];
+  thisMonth: Conversation[];
+  byMonth: Record<string, Conversation[]>;
+};
+
+const ConversationPopoverContent = ({ children }: { children: React.ReactNode }) => (
+  <PopoverContent
+    align="end"
+    alignOffset={-46}
+    side="right"
+    sideOffset={-16}
+    className={cn("bg-[#F5F5F7] w-[100px] border border-[#D7D7D7] shadow-none rounded-[6px] p-3")}
+  >
+    {children}
+  </PopoverContent>
+);
+
 export default function ConversationHistory({ className, tenant }: Props) {
+  const router = useRouter();
   const [conversations, setConversations] = useState<Conversation[]>([]);
   const [isLoading, setIsLoading] = useState(true);
 
+  const fetchConversations = useCallback(async () => {
+    setIsLoading(true);
+    try {
+      const res = await fetch("/api/conversations", { headers: { tenant: tenant.slug } });
+      const json = await res.json();
+      const conversations = conversationListResponseSchema.parse(json);
+      setConversations(conversations);
+    } catch (error) {
+      console.error("Error fetching conversations:", error);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [tenant.slug]);
+
   useEffect(() => {
-    (async () => {
-      setIsLoading(true);
-      try {
-        const res = await fetch("/api/conversations", { headers: { tenant: tenant.slug } });
-        const json = await res.json();
-        const conversations = conversationListResponseSchema.parse(json);
-        setConversations(conversations);
-      } catch (error) {
-        console.error("Error fetching conversations:", error);
-      } finally {
-        setIsLoading(false);
+    fetchConversations();
+  }, [fetchConversations]);
```

Comment:
This is a style thing but I like to include hooks before the component code.  I might miss this when reading through this component.  This is just a little too buried for me.